### PR TITLE
Add examples

### PR DESCRIPTION
**Pull request by gholdman1**
DOC: Add examples to various functions in `scipy.special`, namely `assoc_laguerre`, `jvp`, and yvp`. Partial Fix for these functions (xref #7168).

### CODE REVIEW

File: scipy/special/_basic.py [hunk 1]
```diff
@@ -645,6 +645,24 @@ def jvp(v, z, n=1):
...
            https://dlmf.nist.gov/10.6.E7
 
+    Examples
+    --------
+    Compute the first derivative of the v=2 Bessel Function of the first kind at z=1.
+
+    >>> from scipy import special
+    >>> v=2
+    >>> z=1
+    >>> result = special.jvp(v,z)
```
Comment (ilayn) [inline scipy/special/_basic.py]: nitpick pep8 commas

File: scipy/special/_basic.py [hunk 2]
```diff
@@ -645,6 +645,24 @@ def jvp(v, z, n=1):
...
+    >>> result
+    0.21024361588113258
+
+    Using the recurrence relation found in Formula 10.6.1 in Reference [1]
+    we can perform a sanity check.
+
+    >>> special.jv(v-1,z)-special.jv(v+1,z)
```
Comment (ilayn) [inline scipy/special/_basic.py]: pep8 space after comma

File: scipy/special/_basic.py [hunk 3]
```diff
@@ -685,6 +703,16 @@ def yvp(v, z, n=1):
...
            https://dlmf.nist.gov/10.6.E7
 
+    Examples
+    --------
+    Compute the first derivative of the v=2 Bessel function of the second kind  at z=1.
+
+    >>> from scipy import special
+    >>> v=2
+    >>> z=1
+    >>> result = special.yvp(v,z)
```
Comment (ilayn) [inline scipy/special/_basic.py]: You don't need the intermediate `result` variables. You can directly print the result just as you did with `assoc_laguerre`

File: scipy/special/_basic.py [hunk 4]
```diff
@@ -1064,6 +1092,19 @@ def assoc_laguerre(x, n, k=0.0):
     The polynomial :math:`L^{(k)}_n(x)` is orthogonal over ``[0, inf)``,
     with weighting function ``exp(-x) * x**k`` with ``k > -1``.
 
+    Examples
+    --------
+    Calculate the n=2 Laguerre polynomial at x=1
+    >>> from scipy.special import assoc_laguerre
+    >>> assoc_laguerre(1,2)
```
Comment (ilayn) [inline scipy/special/_basic.py]: pep8 comma

File: scipy/special/_basic.py [hunk 5]
```diff
@@ -1064,6 +1092,19 @@ def assoc_laguerre(x, n, k=0.0):
...
+    -0.5
+
+    Calculate the n=2, k=3 Laguerre polynomial at x=1
+
+    >>> from scipy.special import assoc_laguerre
+    >>> assoc_laguerre(1,2,3)
```
Comment (ilayn) [inline scipy/special/_basic.py]: PEP8 commas

**Review by ilayn — CHANGES_REQUESTED** [review]
Thank you for the effort @gholdman1 ! I've added very minor comments.

**Comment by gholdman1** [comment]
Thank you! I've made the changes. Please let me know if you find anything else.

I'll be adding more examples in this branch soon. Will I need to open a new PR?

**Comment by gholdman1** [comment]
Hello! I tried fixing some errors given by the tests. In the process, I may have messed up the commit history and may not be able to push to `origin/add-examples`. I'll be making a new branch and opening a new PR.

**Pull request closed without merging**